tools/mirror/src/chain_tracker.rs

```diff
@@ -757,17 +841,40 @@ impl TxTracker {
                             // to keep track of which receipts are descendants of our transactions, so that we can reverse any stake actions
                             crate::put_pending_outcome(db, receipt_id, HashSet::default())?;
                         }
+                        if crate::execution_status_good(&outcome.execution_outcome.outcome.status) {
+                            match outcome.receipt.receipt {
+                                ReceiptEnumView::Action { actions, .. } => {
+                                    // since this receipt was recorded in the DB, it means that this stake action
+                                    // resulted from one of our txs (not an external/manual stake action by the test operator),
+                                    // so we want to reverse it.
+                                    for a in actions {
+                                        if let ActionView::Stake { public_key, stake } = a {
+                                            if stake > 0 {
+                                                staked_accounts.insert(
+                                                    (
+                                                        outcome.receipt.receiver_id.clone(),
```

Comment:
This indentation is deep.
(That's a neutral comment, feel free to keep).

Reply:
yea true, quite ugly. refactored it